sidecar/src/unix.rs

```diff
@@ -19,16 +21,112 @@ use tokio::select;
 
 use tokio::net::UnixListener;
 use tokio::sync::mpsc;
+use tokio::sync::mpsc::Receiver;
+use tokio::task::JoinHandle;
 use tokio_util::sync::CancellationToken;
 
 use crate::interface::blocking::TelemetryTransport;
 use crate::interface::TelemetryServer;
 use datadog_ipc::platform::Channel as IpcChannel;
+use ddtelemetry::data::metrics::{MetricNamespace, MetricType};
+use ddtelemetry::metrics::ContextKey;
+use ddtelemetry::worker::{
+    LifecycleAction, TelemetryActions, TelemetryWorkerBuilder, TelemetryWorkerHandle,
+};
 
 use crate::setup::{self, Liaison};
 
 use crate::config::{self, Config};
 
+struct MetricData<'a> {
+    worker: &'a TelemetryWorkerHandle,
+    server: &'a TelemetryServer,
+    submitted_payloads: ContextKey,
+    active_sessions: ContextKey,
+}
+impl<'a> MetricData<'a> {
+    async fn send(&self, key: ContextKey, value: f64) {
+        let _ = self
+            .worker
+            .send_msg(TelemetryActions::AddPoint((value, key, vec![])))
+            .await;
+    }
+
+    async fn collect_and_send(&self) {
+        future::join_all(vec![
+            self.send(
+                self.submitted_payloads,
+                self.server.submitted_payloads.swap(0, Ordering::SeqCst) as f64,
+            ),
+            self.send(
+                self.active_sessions,
+                self.server.active_session_count() as f64,
+            ),
```

Comment:
Eih, no efficiency concerns here, that's just the API I discovered. Can you please elaborate which API exactly I should use instead?

Reply:
for `submitted_payload` since it's a count you could send `1` instead of aggregating values in an atomic variable and sending the value.
For `active_sessions`, you would send the value of the gauge avery time you mutate it

Reply:
It's fine as is, but you wouldn't need the reception task